[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:158 starting continuous writes to the database
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:162 checking whether writes are increasing
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m integration.ha_tests.helpers:helpers.py:242 Initial writes {'testing.postgresql-k8s-0': 144, 'testing.postgresql-k8s-1': 145, 'testing.postgresql-k8s-2': 146}
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m integration.ha_tests.helpers:helpers.py:248 Retry writes {'testing.postgresql-k8s-0': 271, 'testing.postgresql-k8s-1': 272, 'testing.postgresql-k8s-2': 273}
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:165 Get leader unit
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:171 Run pre-upgrade-check action
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:183 Inject dependency fault
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:188 Refresh the charm
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:191 Get first upgrading unit
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:195 Wait for upgrade to fail on first upgrading unit
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:202 Ensure continuous_writes while in failure state on remaining units
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m integration.ha_tests.helpers:helpers.py:242 Initial writes {'testing.postgresql-k8s-0': 2866, 'testing.postgresql-k8s-1': 2868}
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m integration.ha_tests.helpers:helpers.py:248 Retry writes {'testing.postgresql-k8s-0': 3749, 'testing.postgresql-k8s-1': 3751}
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:205 Re-run pre-upgrade-check action
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:209 Re-refresh the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-k8s/0 [idle] waiting: other units upgrading first...
  postgresql-k8s/1 [idle] blocked: ready to rollback application
  postgresql-k8s/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-k8s/0 [idle] waiting: other units upgrading first...
  postgresql-k8s/1 [idle] blocked: ready to rollback application
  postgresql-k8s/2 [idle] maintenance: upgrade completed
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:221 checking whether writes are increasing
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m integration.ha_tests.helpers:helpers.py:242 Initial writes {'testing.postgresql-k8s-0': 13510, 'testing.postgresql-k8s-1': 13512, 'testing.postgresql-k8s-2': 13514}
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m integration.ha_tests.helpers:helpers.py:248 Retry writes {'testing.postgresql-k8s-0': 13643, 'testing.postgresql-k8s-1': 13644, 'testing.postgresql-k8s-2': 13645}
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:224 Resume upgrade
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:228 Wait for application to recover
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-k8s/0 [idle] waiting: other units upgrading first...
  postgresql-k8s/1 [executing] maintenance: stopping charm software
  postgresql-k8s/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-k8s/0 [executing] waiting: other units upgrading first...
  postgresql-k8s/1 [executing] maintenance: upgrading unit
  postgresql-k8s/2 [executing] maintenance: upgrade completed
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-k8s/0 [idle] waiting: other units upgrading first...
  postgresql-k8s/1 [idle] waiting: awaiting for cluster to start
  postgresql-k8s/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-k8s/0 [idle] waiting: other units upgrading first...
  postgresql-k8s/1 [idle] waiting: awaiting for cluster to start
  postgresql-k8s/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-k8s/0 [idle] waiting: other units upgrading first...
  postgresql-k8s/1 [idle] active: 
  postgresql-k8s/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-k8s/0 [executing] maintenance: 
  postgresql-k8s/1 [idle] maintenance: upgrade completed
  postgresql-k8s/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-k8s/0 [idle] waiting: awaiting for cluster to start
  postgresql-k8s/1 [idle] maintenance: upgrade completed
  postgresql-k8s/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-k8s/0 [idle] maintenance: upgrade completed
  postgresql-k8s/1 [idle] maintenance: upgrade completed
  postgresql-k8s/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-k8s/0 [idle] active: 
  postgresql-k8s/1 [idle] active: 
  postgresql-k8s/2 [idle] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-k8s/0 [idle] active: 
  postgresql-k8s/1 [idle] active: 
  postgresql-k8s/2 [idle] active: 
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:234 Ensure continuous_writes after rollback procedure
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m integration.ha_tests.helpers:helpers.py:242 Initial writes {'testing.postgresql-k8s-0': 30402, 'testing.postgresql-k8s-1': 30404, 'testing.postgresql-k8s-2': 30406}
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m integration.ha_tests.helpers:helpers.py:248 Retry writes {'testing.postgresql-k8s-0': 30589, 'testing.postgresql-k8s-1': 30591, 'testing.postgresql-k8s-2': 30592}
[32mINFO    [0m integration.ha_tests.test_upgrade:test_upgrade.py:239 Checking whether no writes were lost
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.18:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m pytest_operator.plugin:plugin.py:936 Model status:

Model    Controller          Cloud/Region        Version  SLA          Timestamp
testing  concierge-microk8s  microk8s/localhost  3.6.4    unsupported  01:57:11Z

App                  Version  Status  Scale  Charm                Channel      Rev  Address         Exposed  Message
postgresql-k8s       14.17    active      3  postgresql-k8s                      2  10.152.183.112  no       
postgresql-test-app           active      1  postgresql-test-app  latest/edge  375  10.152.183.148  no       received database credentials of the first database

Unit                    Workload  Agent  Address       Ports  Message
postgresql-k8s/0        active    idle   10.1.145.213         
postgresql-k8s/1*       active    idle   10.1.145.212         Primary
postgresql-k8s/2        active    idle   10.1.145.211         
postgresql-test-app/0*  active    idle   10.1.145.200         received database credentials of the first database

[32mINFO    [0m pytest_operator.plugin:plugin.py:942 Juju error logs:

unit-postgresql-k8s-2: 01:33:48 ERROR unit.postgresql-k8s/2.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-0: 01:33:48 ERROR unit.postgresql-k8s/0.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-2: 01:48:46 ERROR unit.postgresql-k8s/2.juju-log charms.data_platform_libs.v0.upgrade.VersionError - {'message': 'Versions incompatible, charm 2 can not be upgraded to 15.0', 'cause': 'Upgrades only supported for specific versions, charm versions satisfying requirement ^15', 'resolution': ''}

[32mINFO    [0m pytest_operator.plugin:plugin.py:1024 Forgetting model main...